### PR TITLE
Adds a config option to lock Mutant races behind BYOND Membership

### PR DESCRIPTION
**Pull request by TheAssistant**
This starts off by default.

### CODE REVIEW

**Comment by SconesC** [comment]
I tried to find a greatly enlarged version of the jewish nose caricature on /pol/ but this text will have to do

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -187,7 +187,13 @@ var/global/list/special_roles = list( //keep synced with the defines BE_* in set
 				dat += "<table width='100%'><tr><td width='24%' valign='top'>"
 
 				if(config.mutant_races)
-					dat += "<b>Species:</b><BR><a href='?_src_=prefs;preference=species;task=input'>[pref_species.name]</a><BR>"
+					if(config.paywall_mutant)
+						if(unlock_content)
+							dat += "<b>Species:</b><BR><a href='?_src_=prefs;preference=species;task=input'>[pref_species.name]</a><BR>"
+						else
+							dat += "<b>Species:</b> Human<BR>"
```
Comment (MrStonedOne) [inline code/modules/client/preferences.dm]: Bug, If they already have it set to ligger, and this gets enabled, they will show up as human, but their preference will still be ligger, causing them to automatically change once they buy membership

You should make it selectable, but show some sort of message if paywall_mutant is enabled and they don't have membership. that way they know the option is there, and they can set it, but it does nothing without membership.

Comment (TheAssistant) [inline code/modules/client/preferences.dm]: Alright.

**Comment by Incoming5643** [comment]
Getting dangerously close to paying for in game advantages here.

**Comment by TheAssistant** [comment]
I don't see how this is an ingame advantage. Lizard is purely cosmetic, we removed the bonus sight on them.

**Comment by Boredone** [comment]
If anything, wouldn't it be a disadvantage, as silicons don't see them as human by default?

**Comment by Incoming5643** [comment]
the operating descriptor is "dangerously close"

**Comment by Anonmare** [comment]
Finally I can be the speshul snowflek I was always meant to be

**Comment by WJohn** [comment]
I mean you can "pay for a straight disadvantage" if you want, but we won't have that enabled. I personally really like our current setup where colorful ghosts, OOC color, and more save slots have absolutely 0 effect on gameplay. I think more stuff like that would be nice.

**Comment by Rinoka** [comment]
:-1:

Adding code for buying membership for lizards is silly. I'd understand if you did it for stuff like the cat ears (though lord knows we'd get people still doing it), but lizards are popular on the station and it'll just generate bad blood between coders and parts of the playerbase that aren't necessarily cancerous. Don't do this.

**Comment by GunHog** [comment]
NO. NO NO NO NO NO NO NO NO NO. :-1: x :100: 
Do this for Nekos if you want, but <b>not lizards </b>.

**Comment by Aloraydrel** [comment]
For what purpose

**Comment by freerealestate** [comment]
ours is the fury

**Comment by freerealestate** [comment]
*furry

**Comment by optimumtact** [comment]
:+1: yes if you want to be a furry you can cough up 20 bucks

**Comment by MrStonedOne** [comment]
its literally 1 to 3 bucks a month, depending on how much you buy it in bulk.

**Comment by optimumtact** [comment]
>not buying 6 years
:dollar: :dollar: 
:dollar: :dollar: 
:dollar: :dollar: 
:dollar: :dollar: 
:dollar: :dollar: 
:dollar: :dollar:

**Comment by Falamazeer** [comment]
Does this even do anything useful for our server like offset server costs?
Is there any benefit for anyone besides pushing a byond membership for byonds benefit?

Sorry if I'm unfamiliar, but if it doesn't benefit anyone here I see no reason to hide the feature behind a paywall that you don't control, and I'm unfamiliar with the benefits reaped from having more payers.

**Comment by optimumtact** [comment]
It's so byond doesn't go down the toilet and leave us without a platform to run our game

**Comment by freerealestate** [comment]
Another thought, this would make lizards even more hated than they already
are.

Not that I care that much, I don't play one and the anguish over people who
do is hilarious

**Comment by Falamazeer** [comment]
Byond already shits the bed with broken ad videos. Fuck em. You wanna throw them a bone, make more OOC colors and ghosts or whatever, non-disruptive stuff, which it may seem to be non disruptive, but when lizards get rarer, their effect will be more pronounced.

**Comment by optimumtact** [comment]
maybe we can just sell all the salt it generates

**Comment by Rinoka** [comment]
byond won't tank anytime soon.

besides if it does we can bootleg rig it all so it works, probably

**Comment by Kelenius** [comment]
Where's my downvote button.

I want my downvote button.

Did I break it by pressing it too hard.

**Comment by daman997** [comment]
why :-1:

**Comment by TheAssistant** [comment]
>Does this even do anything useful for our server like offset server costs?

It ensures the ongoing development of BYOND, ensures that we always have a place for SS13, ensures the login servers stay running so we can play, and helps fund new additions to BYOND, which in turn allows us to add new things to SS13.

>Is there any benefit for anyone besides pushing a byond membership for byonds benefit?

When BYOND benefits, we all benefit, players, coders, admins alike. Players can still log in and play the game. Coders can still create content for the game that people will play. Admins can still log in and play the game. I don't quite see why you think BYOND being alive is a bad thing. We can't just jury rig a new BYOND. It's quite frankly impossible without just rewriting BYOND from scratch, which basically ends up the same as all those other failed projects.

>Byond already shits the bed with broken ad videos. Fuck em. 

Maybe if BYOND got enough money in donations to stay afloat they wouldn't need broken ad videos to make money.

>besides if it does we can bootleg rig it all so it works, probably

We really can't. We don't have source access. When BYOND tanks, and it will someday, I highly doubt they will release source. We'd have to recode BYOND from scratch, at which point we might as well just make a new game.

**Comment by freerealestate** [comment]
I don't see this determining BYOND's survival somehow.

**Comment by TheAssistant** [comment]
By itself it doesn't, but every little bit helps.

**Comment by Firecage** [comment]
I disagree with this. Add non disruptive things which won't influence rounds. Instead this is basically adding a payment option which does influence rounds.

**Comment by TheAssistant** [comment]
This doesn't influence rounds. Where the fuck are you getting that retarded nonsense from?

Lizard is purely cosmetic. It had the bonus night vision removed. It's as cosmetic as what hairstyle you're wearing.

**Comment by Firecage** [comment]
And it does influence rounds because you are basically playing to play as another race INGAME. Things like different colour OOC, types of ghosts, byond symbol, etc. doesn't influence rounds at all. Suddenly paying to play as another race does, whether it is cosmetic or not.

**Comment by TheAssistant** [comment]
You failed to explain why being a different race influences the game, because I don't see how lizard gives you any distinct upsides or downsides.

**Comment by Firecage** [comment]
As I said, you play as another race, which DOES influence the round because you can play as another race, doesn't matter if its cosmetic or not. You are paying to play as another race INROUND. I would rather say to add things which doesn't influence the round.

**Comment by TheAssistant** [comment]
Can you stop dodging the question?

I'm asking you WHY playing as another race influences the round? Because it doesn't.

**Comment by Kelenius** [comment]
4. Only lizards are human.

**Comment by Dorsisdwarf** [comment]
You don't even need a hacked module, the Asimov module can have the subject rewritten

**Comment by GunHog** [comment]
What race you play has a very real affect on your round, and not just the Enforce Human Authority config option. Lizards are treated differently ingame, your experience as a lizard will be different as a human, in the same job. Aside from the job restrictions, and that silicons consider you non-human in critical situations (it can kill you for breaking into its chamber, and may harm a lizard to protect a human antagonist), there is a psychological aspect in being a lizard player that will alter your interactions with other players.

Player interaction is one of the most important focuses of the game, and being a different race changes your interactions with other players, *especially* those who choose to play only humans. This obviously can and does influence rounds, on a direct, player-to-player basis.

**Comment by AnturK** [comment]
Not a fan, RP player interaction is the biggest part of the game and race influences that greatly. Not even starting on how this will ostracize lizard players.

(Also i'd be more inclined to listen to this if the author was not hiding behind a sock-puppet account.)

**Comment by bustygoku** [comment]
WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS WHITELIST LIZARDS

**Comment by Falamazeer** [comment]
You are awfully pushy about people ignoring you baseless claims assistant, considering you quoted my words EXCEPTING the most relevant part.

"it may seem to be non disruptive, but when lizards get rarer, their effect will be more pronounced."

**Comment by Miauw62** [comment]
it makes the snowflake liggers mad :+1:

**Comment by d3athrow** [comment]
@TheAssistant Why would you be giving money to a guy who reneged on tom's promise on opensourcing parts of byond after 507 went stable?

**Comment by SconesC** [comment]
>This starts off by default.

**Comment by Incoming5643** [comment]
Stoned already said in game that he would consider turning this on i fit was added.

**Comment by TheNightingale** [comment]
I get that this is a joke PR, but what's the point? What good does this bring to /tg/station at all?

**Comment by TheAssistant** [comment]
>I get that this is a joke PR, but what's the point? What good does this bring to /tg/station at all?

This isn't a joke PR.

>Why would you be giving money to a guy who reneged on tom's promise on opensourcing parts of byond after 507 went stable?

Maybe if they have money they'll hire a competent programmer?
Plus, Tom was probably talking out of his ass to begin with. I mean, the guy tried his hardest to pretend we didn't exist.

>it may seem to be non disruptive, but when lizards get rarer, their effect will be more pronounced.

That's entirely player mindset, and I can't adjust or change that.

>$%#^@. Only lizards are human. 

I could just as easily put in "Only walking talking animatronic vaginas are human.". Your point is?

>What race you play has a very real affect on your round, and not just the Enforce Human Authority config option. 

Once again, this is player mindset. I can't change that. Be the change you want to be.


>Lizards are treated differently ingame, your experience as a lizard will be different as a human, in the same job. 

This is both Server Policy(which I as a coder cannot dictate) and player mindset. I can't change either of those. Try lobbying your local congressman aka the host and headmins for a change.

>Aside from the job restrictions, and that silicons consider you non-human in critical situations (The AI can kill you for breaking into its chamber, and borgs may harm a lizard to protect a human antagonist)

Job restrictions is Server Policy, which I cannot dictate. Try lobbying your local congressman aka the host and headmins for a change.
Silicon treatment of lizards as non human is both Server Policy and Silicon Policy, which I cannot dictate either of the two. Try lobbying your local congressman aka the host and headmins for a change.

>there is a psychological aspect in being a lizard player that will alter your interactions with other players.

There shouldn't be. In my eyes, Lizard should really just be a reskin of human. It's just another player. I don't give a shit about what race they are, all I care about is "is he valid".

**Comment by TheNightingale** [comment]
So... what's the point? What does this achieve?

**Comment by optimumtact** [comment]
more options for downstream branches

**Comment by TheNightingale** [comment]
I'm still having trouble seeing how this benefits us (as in /tg/station) specifically, though - we're not a downstream branch. How does this PR improve, maintain or repair anything?

**Comment by optimumtact** [comment]
gives the headmins more policy tools should they need them

**Comment by TheNightingale** [comment]
In that case, it sounds like a 'just in case' config option (and, arguably, a waste of code) that we wouldn't need in the first place. At the risk of slipping on this slippery slope, we might as well make an option to turn every ID sprite into a pony - it's a policy tool for downstream servers, r-right?

:-1: for now, because we don't need it and it serves no practical purpose whatsoever. (Besides, I like my lizards, thank you very much! :crocodile:

**Comment by TheAssistant** [comment]
>a waste of code

Code is not limited. There is infinite code. Fuck off with this shit.

> At the risk of slipping on this slippery slope, we might as well make an option to turn every ID sprite into a pony - it's a policy tool for downstream servers, r-right?

You're just being retarded at this point.

**Comment by Firecage** [comment]
Waste of code.

**Comment by TheAssistant** [comment]
>Waste of code.

Please explain how I can waste an infinite resource.

**Comment by Aloraydrel** [comment]
I just want to know why would you need this "Just in case". People can donate to byond if they wanted. So why put a race some people enjoy playing and put it behind a paywall?  All I see coming from this if it was active is upsetting players who enjoy playing lizards.

**Comment by Cheridan** [comment]
@d3athrow I don't really think that's fair to Lummox. He's done a lot more to cooperate and help SS13 as a whole than the other Byond devs, who yeah as TheAssistant said tended to treat the platform's most popular game as a secret shame.

**Comment by Firecage** [comment]
PS, I kinda feel like this needs a thread on the forum, especially due to being like super controversial.

**Comment by SconesC** [comment]
Why does this need a thread? It /starts off by default/

Enabling it, now that would be thread-worthy

**Comment by Firecage** [comment]
Scones, except the fact that it was stated that it probably will be turned on if merged?

**Comment by SconesC** [comment]
Enabling it is threadworthy

Merging it as an OPTION for /tg/ and downstreams is such a non-issue I don't see why people are making such a fuss over it

**Comment by Incoming5643** [comment]
because stopping a pull before it gets added to the codebase is a much more realistic goal than either trying to get a revert or convincing the right people to turn the config option off.

**Comment by GunHog** [comment]
It should not be an option at all. Also, we, as the playerbase, have more chances to stop a PR now, than it becoming a config option we may have to convince people to change, possibly several times. Halting it before it becomes a feature may set a precedent that kills this issue for good.

**Comment by Incoming5643** [comment]
On a personal level it burns a bit that someone wants to turn the work me and wjohn did on lizards into a paywall feature that very few people will probably use.

**Comment by Summoners** [comment]
:-1: It shouldn't be an option to even be possible to put the game behind a pay wall. Sure the lizards are just for looks but people play lizards differently and it takes out a gameplay feature that only the few with cash could enjoy.
:-1: :-1: :-1:

**Comment by Shadowlight213** [comment]
>TheAssistant doesnt have BYOND membership

>Pushes to lock things off so BYOND gets more cash

Hypocrite detected

**Comment by SconesC** [comment]
LIDF out in full force

**Comment by Iamgoofball** [comment]
this PR is fuckin hillarious

topkek/10

@Incoming5643 

something that would be cool is if we added tattoos and piercings and shit so the system isn't unused

**Comment by Rinoka** [comment]
@TheAssistant: BYOND staff shill detected

firing the :-1: cannons

but seriously: this is not your code to fuck about with paywalling, especially without putting up a thread about it for actual discussion about the config. ESPECIALLY CONSIDERING nobody playing this game gives a single shit about BYOND, and the chances it'll go under are probably handled if like 5 people pay for subs a month, considering they solely offer the hub for playerkeys to connect to games - not for the game space itself.

**Comment by optimumtact** [comment]
The bad attitude towards byond here is frankly unacceptable

**Comment by SconesC** [comment]
>The bad attitude towards byond here is frankly unacceptable

**Comment by CorruptComputer** [comment]
>>The bad attitude towards byond here is frankly unacceptable

**Comment by Iamgoofball** [comment]
>>>The bad attitude towards byond here is frankly unacceptable

**Comment by Aloraydrel** [comment]
Oy vey!

**Comment by Incoming5643** [comment]
Listen if byond devs came in here and said the service was dying, if they begged us to try and get them some cash, if they told us it was do or die time then maybe I'd give a fuck and be on board with something like this. But they don't, so I don't see why the fuck we have to intentionally gimp our own features just to support them.

Paywall new shit if you must, but taking old shit and paywalling it after everyone is already used to it is ridiculous.

**Comment by Firecage** [comment]
Incoming has actually made the best point so far.

**Comment by Lumipharon** [comment]
It wasn't even so long ago that lizards were admin only, jesus christ this whole pr.

**Comment by Shadowlight213** [comment]
Aye. What incoming said.
Also, lizards have been in for some time, and taking an oldish feature, that a lot of balancing and coding went into, and locking it behind a paywall to a few people is dumb. BYOND isn't dying, there's no reason to lock up features.

**Comment by Lumipharon** [comment]
It's not a feature though, and it has had no balancing involved.
It's mechanically identical to a human player.

The only differences is player behavior, and the fact that the wording of the default silicon lawset excludes them.

**Comment by GunHog** [comment]
They are banned from head roles, so they are mostly, but not fully, identical.

**Comment by Intigracy** [comment]
>>Scones in here shilling saying "MERGE IT NOW IT'S JUST A CONFIG OPTION GUYS" when MSO has clearly stated he'd be turning it on

10/10 enabling shitty behavior.

Have you ever tried to disguise yourself as a ligger after killing them as an antagonist? Ligger players do have benefits to playing as liggers over humans, such as less likely to be killed for a disguise due to the difficulty of pulling it off, and they do have negatives (like silicon treatment).

Though I'm almost positive this is a troll PR because your only response to people's complaints is "a bloo bloo bloo I'm changing the code, if you have issues whine to admins/headmins to change the policy around it"

>Listen if byond devs came in here and said the service was dying, if they begged us to try and get them some cash, if they told us it was do or die time then maybe I'd give a fuck and be on board with something like this. But they don't, so I don't see why the fuck we have to intentionally gimp our own features just to support them.

>Paywall new shit if you must, but taking old shit and paywalling it after everyone is already used to it is ridiculous.

**Comment by Shadowlight213** [comment]
I think Aran was on to something with that comment.

**Comment by Miauw62** [comment]
no matter how much ligger salt this causes and now matter how much i like that, this is a dumb idea.

**Comment by Aranclanos** [comment]
cmon dude you didn't even switch your github account, you gotta try harder

**Comment by Aranclanos** [comment]
have another donger:
8=========D

**Comment by optimumtact** [comment]
good riddance

**Comment by Intigracy** [comment]
8=========D

**Comment by Incoming5643** [comment]
7/10 got me a little salty about spessmen

**Pull request closed without merging**